content/reader.go

```diff
@@ -120,7 +120,10 @@ func ReadAll(r io.Reader, desc ocispec.Descriptor) ([]byte, error) {
 	buf := make([]byte, desc.Size)
 
 	vr := NewVerifyReader(r, desc)
-	if _, err := io.ReadFull(vr, buf); err != nil {
+	if n, err := io.ReadFull(vr, buf); err != nil {
+		if errors.Is(err, io.ErrUnexpectedEOF) {
+			return nil, fmt.Errorf("read failed, expected content size of %d, got %d, for digest %s: %w", desc.Size, n, desc.Digest.String(), err)
```

Comment:
nit: just be consistent with the generic error message of `"read failed: %w"` at line 127.
```suggestion
			return nil, fmt.Errorf("read failed: expected content size of %d, got %d, for digest %s: %w", desc.Size, n, desc.Digest.String(), err)
```

Reply:
#432 asked for something more specific. The core goal of this PR is to provide more info than line 127 does. Maybe I went overboard with the extra info, but I expect I'd want all three pieces of data if I was trying to fix the problem with my content triggering this behavior.

Reply:
The error message looks useful!
+1 on changing `read failed,` to `read failed:`